lib/util/Queue.js

```diff
@@ -47,7 +42,7 @@ class Queue {
 	 * @returns {T | undefined} The head of the queue of `undefined` if this queue is empty.
 	 */
 	dequeue() {
-		const result = this._iterator.next();
+		const result = this._set[Symbol.iterator]().next();
```

Comment:
It'd be better to cache the result of this call, and reset the cached iterator when result.done === true and `this._set.size > 0`. Otherwise you're spending a lot of time unnecessarily regenerating the iterator.

Reply:
@dmichon-msft Good catch, do you want to send pr?

Reply:
https://github.com/webpack/webpack/pull/19086

@alexander-akait @dmichon-msft can you guys check this?